docs/manuals/gui/viselements/charts/heatmap.md

```diff
@@ -0,0 +1,223 @@
+```py
+from taipy.gui import Gui
+import pandas as pd
+
+data = [
+    pd.DataFrame({
+        'harvest': [
+            [0.8, 2.4, 2.5, 3.9, 0.0, 4.0, 0.0],
+            [2.4, 0.0, 4.0, 1.0, 2.7, 0.0, 0.0],
+            [1.1, 2.4, 0.8, 4.3, 1.9, 4.4, 0.0],
+            [0.6, 0.0, 0.3, 0.0, 3.1, 0.0, 0.0],
+            [0.7, 1.7, 0.6, 2.6, 2.2, 6.2, 0.0],
+            [1.3, 1.2, 0.0, 0.0, 0.0, 3.2, 5.1],
+            [0.1, 2.0, 0.0, 1.4, 0.0, 1.9, 6.3]
+        ],
+        'vegetables': ["cucumber",
+                       "tomato",
+                       "lettuce",
+                       "asparagus",
+                       "potato",
+                       "wheat",
+                       "barley"]
+    }),
+    pd.DataFrame({
+        'farmers': ["Farmer Joe",
+                    "Upland Bros.",
+                    "Smith Gardening",
+                    "Agrifun",
+                    "Organiculture",
+                    "BioGoods Ltd.",
+                    "Cornylee Corp."],
+    })
+]
+options = {'colorscale': 'Rainbow'}
+md = """
+## Heatmap with Categorical Axis Labels
+<|{data}|chart|type=heatmap|z=0/harvest|x=1/farmers|y=0/vegetables|options={options}|>
+"""
+
+Gui(md).run()
+```
+![Heatmap with Categorical Axis Labels](heatmap_axis_labels.png)
+
+### Annotated Heatmap
+In this chart, we have average temperature (°F) in seasons of top 4 US cities
+```py
+from taipy.gui import Gui
+import pandas as pd
+
+seasons = ['Winter', 'Summer', 'Spring', 'Autumn']
+
+cities = ['New York', 'Los Angeles', 'Chicago', 'Houston']
+
+temperature = [
+    [33.7, 74.49, 57.8, 57.6],
+    [60.1, 68.2, 64.5, 65.7],
+    [22.89, 72.2, 55.7, 51.6],
+    [53.0, 83.3, 72.7, 53.0],
+]
+
+data = [
+    pd.DataFrame({
+        'temperature': temperature,
+        'cities': cities
+    }),
+    pd.DataFrame({
+        'seasons': seasons,
+    })
+]
+
+options = {'colorscale': 'Portland'}
+
+layout = {
+    'annotations': [],
+    'xaxis': {
+        'ticks': '',
+        'side': 'top'
+    },
+    'yaxis': {
+        'ticks': '',
+        'ticksuffix': ' ',
+        'width': 700,
+        'height': 700,
+        'autosize': False
+    }
+}
+
+for i in range(len(cities)):
+  for j in range(len(seasons)):
+    currentValue = temperature[i][j]
+    if (currentValue < 40 or currentValue > 70):
+        textColor = 'white'
+    else:
+        textColor = 'black'
+    result = {
+        'xref': 'x1',
+        'yref': 'y1',
+        'x': seasons[j],
+        'y': cities[i],
+        'text': str(temperature[i][j]) + '°F',
+        'font': {
+            'family': 'Arial',
+            'size': 12,
+            'color': textColor
+        },
+        'showarrow': False
+    }
+    layout['annotations'].append(result)
+
+md = """
+## Annotated Heatmap
+<|{data}|chart|type=heatmap|z=0/temperature|x=1/seasons|y=0/cities|options={options}|layout={layout}|>
+"""
+
+Gui(md).run()
+```
+![Annotated Heatmap](heatmap_annotated.png)
+
+### Heatmap with Unequal Block Sizes
```

Comment:
Ok so that's a great example copied from Plotly.
Let's keep it that way for he time being.
Fortunately, the CS team will come up with a different use case.

If we keep this, we will want to point to the Plotly doc though. For respect.